pkg/network/usm/ebpf_main.go

```diff
@@ -323,6 +324,11 @@ func (e *ebpfProgram) init(buf bytecode.AssetReader, options manager.Options) er
 		manager.ConstantEditor{Name: "ephemeral_range_begin", Value: uint64(begin)},
 		manager.ConstantEditor{Name: "ephemeral_range_end", Value: uint64(end)})
 
+	// Enable event flushes from socket filter programs if possible
+	kversion, err := kernel.HostVersion()
+	isDirectFlushSupported := err == nil && kversion >= kernel.VersionCode(5, 4, 0)
```

Comment:
Let's cache the kernel version in the initialization of the eBPF program, or at least in `Init` rather than `init` (which can be called up to 3 times)